pkgs/development/libraries/v8/default.nix

```diff
@@ -93,16 +106,25 @@ stdenv.mkDerivation rec {
     "treat_warnings_as_errors=false"
     "v8_enable_i18n_support=true"
     "use_gold=false"
-    "use_system_xcode=true"
+    "init_stack_vars=false"
     # ''custom_toolchain="//build/toolchain/linux/unbundle:default"''
     ''host_toolchain="//build/toolchain/linux/unbundle:default"''
     ''v8_snapshot_toolchain="//build/toolchain/linux/unbundle:default"''
   ] ++ lib.optional stdenv.cc.isClang ''clang_base_path="${stdenv.cc}"'';
 
   NIX_CFLAGS_COMPILE = "-O2";
-
-  nativeBuildInputs = [ gn ninja pkg-config python ]
-    ++ lib.optionals stdenv.isDarwin [ xcbuild darwin.DarwinTools ];
+  FORCE_MAC_SDK_MIN = stdenv.targetPlatform.sdkVer or "10.12";
+
+  nativeBuildInputs = [
+    gn
+    ninja
+    pkg-config
+    python
+  ] ++ lib.optionals stdenv.isDarwin [
+    xcbuild
+    darwin.DarwinTools
```

Comment:
```suggestion
    DarwinTools
```

Reply:
That doesn't work, since `DarwinTools` isn't in scope and not available outside `darwin`. To make it work, the arguments have to be edited and `DarwinTools` has to be supplied manually in `all-packages.nix`.

Reply:
You are right. We should always use the smallest possible scope though and darwin should never be an input in an package like pkgs is not.

Reply:
`pkgs` is a bit different, since `callPackage` automatically supplies automatically supplies the required inputs from it. The same isn't true for package sets like `darwin`, `pythonPackages`, `nodePackages`, etc, hence why they frequently appear as package arguments. The less trickery done in `all-packages.nix`, the better, imo - it's confusing when only viewing the package file and, at least when importing packages from a set, requires double updates when inputs change.